esphome/components/beken_spi_led_strip/light.py

```diff
@@ -0,0 +1,133 @@
+from dataclasses import dataclass
+
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome import pins
+from esphome.components import libretiny, light
+from esphome.const import (
+    CONF_CHIPSET,
+    CONF_IS_RGBW,
+    CONF_MAX_REFRESH_RATE,
+    CONF_NUM_LEDS,
+    CONF_OUTPUT_ID,
+    CONF_PIN,
+    CONF_RGB_ORDER,
+)
+
+CODEOWNERS = ["@Mat931"]
+DEPENDENCIES = ["libretiny"]
+
+beken_spi_led_strip_ns = cg.esphome_ns.namespace("beken_spi_led_strip")
+BekenSPILEDStripLightOutput = beken_spi_led_strip_ns.class_(
+    "BekenSPILEDStripLightOutput", light.AddressableLight
+)
+
+RGBOrder = beken_spi_led_strip_ns.enum("RGBOrder")
+
+RGB_ORDERS = {
+    "RGB": RGBOrder.ORDER_RGB,
+    "RBG": RGBOrder.ORDER_RBG,
+    "GRB": RGBOrder.ORDER_GRB,
+    "GBR": RGBOrder.ORDER_GBR,
+    "BGR": RGBOrder.ORDER_BGR,
+    "BRG": RGBOrder.ORDER_BRG,
+}
+
+
+@dataclass
+class LEDStripTimings:
+    bit0: int
+    bit1: int
+    spi_frequency: int
+
+
+CHIPSETS = {
+    "WS2812": LEDStripTimings(
+        0b11100000, 0b11111100, 6666666
+    ),  # Clock divider: 9, Bit time: 1350ns
+    "SK6812": LEDStripTimings(
+        0b11000000, 0b11111000, 7500000
+    ),  # Clock divider: 8, Bit time: 1200ns
+    "APA106": LEDStripTimings(
+        0b11000000, 0b11111110, 5454545
+    ),  # Clock divider: 11, Bit time: 1650ns
+    "SM16703": LEDStripTimings(
+        0b11000000, 0b11111110, 7500000
+    ),  # Clock divider: 8, Bit time: 1200ns
+}
+
+
+CONF_IS_WRGB = "is_wrgb"
+
+SUPPORTED_PINS = {
+    libretiny.const.FAMILY_BK7231N: [16],
+    libretiny.const.FAMILY_BK7231T: [16],
+    libretiny.const.FAMILY_BK7251: [16],
+}
+
+
+def _validate_pin(value):
+    family = libretiny.get_libretiny_family()
+    if family not in SUPPORTED_PINS:
+        raise cv.Invalid(f"Chip family {family} is not supported.")
+    if value not in SUPPORTED_PINS[family]:
+        supported_pin_info = ", ".join(f"{x}" for x in SUPPORTED_PINS[family])
+        raise cv.Invalid(
+            f"Pin {value} is not supported on the {family}. Supported pins: {supported_pin_info}"
+        )
+    return value
+
+
+def _validate_num_leds(value):
+    max_num_leds = 165  # 170
+    if value.get(CONF_IS_RGBW, False) or value.get(CONF_IS_WRGB, False):
```

Comment:
Since there is a default of False for both of these, you can just access it directly
```suggestion
    if value[CONF_IS_RGBW] or value[CONF_IS_WRGB]:
```